vyper/semantics/analysis/utils.py

```diff
@@ -645,15 +645,14 @@ def check_modifiability(node: vy_ast.VyperNode, modifiability: Modifiability) ->
         return all(check_modifiability(item, modifiability) for item in node.elements)
 
     if isinstance(node, vy_ast.Call):
-        args = node.args
-        if len(args) == 1 and isinstance(args[0], vy_ast.Dict):
-            return all(check_modifiability(v, modifiability) for v in args[0].values)
-
         call_type = get_exact_type_from_node(node.func)
 
-        # builtins
-        call_type_modifiability = getattr(call_type, "_modifiability", Modifiability.MODIFIABLE)
-        return call_type_modifiability >= modifiability
+        # structs and interfaces
+        if isinstance(call_type, TYPE_T):
```

Comment:
it's more complicated but i think this actually should use `TYPE_T.check_modifiability_for_call` (which can dispatch to something like `StructT.ctor_modifiability_for_call`). the way you have it written, a struct instance can pass the modifiability check.